aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -140,6 +140,41 @@ bool is_nonzero(const Tensor& self) {
   TORCH_INTERNAL_ASSERT(false, "Expected non-Tensor backend scalar");
 }
 
+namespace {
+
+inline at::Tensor scalar_to_tensor_default_dtype(
```

Comment:
Better add a comment explaining what this function is for

Reply:
Originally I had a few suggestions for how to modernize this function, but now I have a different idea: what about copying and putting it next to "scalar_to_tensor?":

https://github.com/pytorch/pytorch/blob/2b02d154dba6ec659f6d501feefcd67044033ed2/aten/src/ATen/ScalarOps.h#L12

Except the dtypes would be different, as you're acquiring them.

Reply:
Have moved the function and added some comments.

Please review.